modules/widget-visibility/widget-conditions.php

```diff
@@ -110,6 +110,15 @@ public static function widget_conditions_options_echo( $major = '', $minor = ''
 				else if ( 'post' == $minor )
 					$minor = 'post_type-post';
 
+				$post_types = get_post_types( array( 'public' => true ), 'objects' );
+
+				$support_archive_page = array();
+				foreach ( $post_types as $post_type ) {
+					if( $post_type->has_archive ) {
```

Comment:
@atrus1701 you can actually check `'has_archive'` in `get_post_types` and avoid this loop.